### DISCUSSION TITLE
[Feature] Option to only show images without any faces

### DISCUSSION THREAD

**Discussion by sebbi08** [damongolding/immich-kiosk#107] [Feature request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Is it possible to add this feature to only show images where there are no faces detected ?

**Comment by damongolding** [comment]
This feels like a very niche request. May I enquire as to why you would want such an option?

**Comment by sebbi08** [comment]
Currently, I maintain an album for my kiosk display and I notice that I put all the images in there without people in there, so it would be easier to just hide all without a face. So only some convenience for me.

**Comment by jedney** [comment]
I would really like the opposite of this (only displaying photo's with people).

Would it maybe be possible for the person include/exclude parameter to take a wildcard? 

e.g. excluded_people="\*" for @sebbi08 (or in my case person="\*")

p.s. This is a fabulous app BTW. Thank you for all your hard work @damongolding

**Reply by damongolding** [reply]
I assume you would only want known people @jedney ?

**Reply by jedney** [reply]
Thanks for the quick response. 

In a perfect world, yes, only people that are named in Immich... 

That said, if that's too tricky, there is still some benefit in only showing photo's with _any_ persons/faces.

My use case is to keep my photo frame showing people - rather than landscapes, animals, plates of food, app screenshots, and other guff that finds its way into my photo collection :-)

**Reply by damongolding** [reply]
My initial thought would be to use `person=all` like how `album=all` does. Then that way `excluded_people` would still work 🤔

**Reply by jedney** [reply]
That would be ideal. I will keep my fingers crossed.

I'm mindful i somewhat hijacked @sebbi08 's request with my request... if you do build this feature, I wonder if you could maybe do an inverse option of `people=none` as part of that enhancement to cover people with the opposite requirement (like @sebbi08 )?

Thanks for considering this.

**Reply by damongolding** [reply]
@jedney I have started implementation in the image `ghcr.io/damongolding/immich-kiosk-development:0.19.1-beta.3` if you would like to try it and give feedback `person=all` with enable it.

**Reply by jedney** [reply]
Thanks @damongolding . I will give it a try over the weekend. Thanks for the quick turn-around.

**Reply by jedney** [reply]
Looks good @damongolding . I've left it running for a while now and so far I haven't noticed any pictures without named persons.

Thanks heaps for building this, it's really appreciated.

**Comment by pravardhanreddy** [comment]
Hi @damongolding I have the same use case as @sebbi08 

I live with a roommate, and I set up a kiosk in the hall. It would be weird to see my family members' faces on there. So if it's possible to only show images without any faces in them, that would be ideal for me :) Thanks again for the wonderful project

**Comment by damongolding** [comment]
Added the `filter_exclude_faces` in 0.38.0
